### PR TITLE
Integrate workload identity

### PR DESCRIPTION
**Pull request by SulakshanaBandara**
Description:

- Update the AKS (Azure Kubernetes Service) cluster and configuring federated identities. The goal is to integrate Azure AD Workload Identity.

Changes made:

- Integrate a new resource as Federated_identity_credentials.
- Updated the AKS cluster by enabling workload identity in the AKS cluster
- Updated the AKS cluster by enabling OIDC url in the AKS cluster.

Related GIT issue:
[https://github.com/wso2-enterprise/iam-engineering/issues/863](https://github.com/wso2-enterprise/iam-engineering/issues/863)

### CODE REVIEW

File: modules/azurerm/Federated-Identity/variables.tf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  description = "OIDC issuer url"
+  type        = string
+}
+
+variable "federated_audience" {
+  description = "federated credential audience"
+  type        = list(any)
+}
+
+variable "workload_sa_name" {
```
Comment (ThilinaManamgoda) [inline modules/azurerm/Federated-Identity/variables.tf:37]: Shall we stick to `subject` as this module is generic. Value passed for the subject can be service account or else that is up to the end user.

File: modules/azurerm/Federated-Identity/variables.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  type        = string
+}
+
+variable "oidc_issuer_url" {
+  description = "OIDC issuer url"
+  type        = string
+}
+
+variable "federated_audience" {
+  description = "federated credential audience"
```
Comment (ThilinaManamgoda) [inline modules/azurerm/Federated-Identity/variables.tf:33]: Let's stick to `audience` with description `Audience for this Federated Identity Credential`

File: modules/azurerm/Federated-Identity/variables.tf [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  type        = list(any)
```
Comment (ThilinaManamgoda) [inline modules/azurerm/Federated-Identity/variables.tf:34]: Shouldn't it be `list(string)`

File: modules/azurerm/Federated-Identity/variables.tf [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+#
+# --------------------------------------------------------------------------------------
+
+variable "resource_group_name" {
+  description = "Resource group name"
+  type        = string
+}
+
+variable "federated_identity_name" {
+  description = "AKS cluster name"
```
Comment (ThilinaManamgoda) [inline modules/azurerm/Federated-Identity/variables.tf:18]: Let's stick to `name` and generic description `Name of this Federated Identity Credential`.

Comment (ThilinaManamgoda) [reply modules/azurerm/Federated-Identity/variables.tf:18]: Have we looked what is the abbreviation for  Federated Identity Credential ?

File: modules/azurerm/Federated-Identity/versions.tf [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+# You may not alter or remove any copyright or other notice from copies of this content.
+#
+# --------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.13"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.0.0"
```
Comment (ThilinaManamgoda) [inline modules/azurerm/Federated-Identity/versions.tf:17]: Does this support `>= 3.52.0` ?

**Review by ThilinaManamgoda — CHANGES_REQUESTED** [review]

**Review by ThilinaManamgoda — APPROVED** [review]

File: .idea/$PROJECT_FILE$ [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ThilinaManamgoda) [inline .idea/$PROJECT_FILE$:1]: Let's remove this file

**Review by ThilinaManamgoda — CHANGES_REQUESTED** [review]

**Review by ThilinaManamgoda — APPROVED** [review]

**Review by Migara-Pramod — APPROVED** [review]

**Pull request merged**